docs: announcement for the new SQL Editor (#727)
bytebase/bytebase.com: content/blog/the-sql-editor-for-developers-and-dbas.md

- [Database-based access control](#database-based-access-control): All databases in the protected environments can not be queried by developers in SQL Editor. DBAs can configure a database allowlist to grant developers the ability to query some of databases.
- [Data anonymization](#data-anonymization): The query result of sensitive columns will be masked when developers run a query in SQL Editor. It is applicable to all query types, such as subquery, JOIN, Common Table Expression (CTE).
- [Collaborative sheets](#collaborative-sheets): DBAs and developers can save SQL statements as sheets to share with others in SQL Editor. Moreover, this release supports saving the uploaded SQL scripts as sheets to be used in issues to apply database changes, a single sheet can hold up to 100MB sized SQL statements.
- Good user experience: This release presents SQL Editor with a fresh look and improves the code auto-completion.

### Admin Mode

When users want to connect to databases to run admin commands, they can do it in SQL Editor Admin mode with a DBA account like the screenshot below. You can take a look at the document [Admin Mode](/docs/sql-editor/admin-mode) to learn more details.

> ![admin-mode](/static/blog/the-sql-editor-for-developers-and-dbasadmin-mode.webp)

Reviewer: ```suggestion
![admin-mode](/static/blog/the-sql-editor-for-developers-and-dbas/admin-mode.webp)
```

### Database-based Access Control

> When users want to allow developers to access some databases in the production environment, they can mark the production environment as a protected environment and configure an allowlist. You can get more details from the document [Database Access Control](https://www.bytebase.com/docs/administration/database-access-control).

Reviewer: ```suggestion
When users want to allow developers to access some databases in the production environment, they can mark the production environment as a protected environment and configure an allowlist. You can get more details from the document [Database Access Control](/docs/administration/database-access-control).
```

> ![access-control](/static/blog/the-sql-editor-for-developers-and-dbasaccess-control.webp)

Reviewer: ```suggestion
![access-control](/static/blog/the-sql-editor-for-developers-and-dbas/access-control.webp)
```

### Data Anonymization

> When developers query tables containing sensitive columns in SQL Editor, the result of these columns will display as "**\*\***". You can learn the configuration details from the document [Anonymize Data](https://www.bytebase.com/docs/administration/anonymize-data).

Reviewer: ```suggestion
When developers query tables containing sensitive columns in SQL Editor, the result of these columns will display as "**\*\***". You can learn the configuration details from the document [Anonymize Data](/docs/administration/anonymize-data).
```

> ![anonymize-data](/static/blog/the-sql-editor-for-developers-and-dbasanonymize-data.webp)

Reviewer: ```suggestion
![anonymize-data](/static/blog/the-sql-editor-for-developers-and-dbas/anonymize-data.webp)
```

### Collaborative Sheets

> For commonly-used SQL statements, you can save them as sheets and set up a public SQL script library in your team. For large SQL scripts (up to 100M), you can use them in issues after uploading them as sheets (see the screenshot below). You can explore the document [Manage SQL Scripts with Sheet](https://www.bytebase.com/docs/sql-editor/manage-sql-scripts) to get additional details.

Reviewer: ```suggestion
For commonly-used SQL statements, you can save them as sheets and set up a public SQL script library in your team. For large SQL scripts (up to 100M), you can use them in issues after uploading them as sheets (see the screenshot below). You can explore the document [Manage SQL Scripts with Sheet](/docs/sql-editor/manage-sql-scripts) to get additional details.
```

> ![large-script](/static/blog/the-sql-editor-for-developers-and-dbaslarge-script.webp)

Reviewer: ```suggestion
![large-script](/static/blog/the-sql-editor-for-developers-and-dbas/large-script.webp)
```

### Data Anonymization

When developers query tables containing sensitive columns in SQL Editor, the result of these columns will display as "**\*\***". You can learn the configuration details from the document [Anonymize Data](/docs/administration/anonymize-data).
![anonymize-data](/static/blog/the-sql-editor-for-developers-and-dbas/anonymize-data.webp)

### Audit Logs

DBAs can browse all queries executed in SQL Editor with the audit logs. You can check out the document [Audit Log](/docs/administration/audit-log) to get more details.
![audit-log](/static/blog/the-sql-editor-for-developers-and-dbas/audit-log.webp)

### Collaborative Sheets

For commonly-used SQL statements, you can save them as sheets and set up a public SQL script library in your team. For large SQL scripts (up to 100M), you can use them in issues after uploading them as sheets (see the screenshot below). You can explore the document [Manage SQL Scripts with Sheet](/docs/sql-editor/manage-sql-scripts) to get additional details.
![large-script](/static/blog/the-sql-editor-for-developers-and-dbas/large-script.webp)

## Try it Out

> With the new SQL Editor in Bytebase Console, you can run database admin commands in Admin mode, anonymize sensitive data when executing queries, configure a database allowlist to manage database access permission, and save commonly-used SQL scripts to collaborate with team members. Besides that, you can view all queries in SQL Editor with the audit logs. We hope you can find the SQL Editor to be a useful tool.

Reviewer: ```suggestion
With the new SQL Editor in Bytebase Console, you can run database admin commands in Admin mode, anonymize sensitive data when fetching query results, configure a database allowlist to manage database access permission, and save commonly-used SQL scripts to collaborate with team members. Besides that, you can inspect all queries in SQL Editor via audit logs.
```

> If you have any comments or questions, don’t hesitate to let us know. You can join our discord at https://discord.gg/Fac9nmZ95j to learn more about Bytebase.

Reviewer: ```suggestion
If you have any comments or questions, don’t hesitate to let us know. You can join our [discord](https://discord.gg/Fac9nmZ95j) to learn more about Bytebase.
```
